File: arm.64/langs/python/src/find.py
import os
import sys
from pathlib import Path
from typing import List, Optional

def find(target_dir: Optional[str] = None) -> List[str]:
    if target_dir:
        d = Path(target_dir)
    elif os.environ.get("DIST_DIR"):
        d = Path(os.environ["DIST_DIR"])
    else:
        exe = Path(sys.argv[0]).resolve()
        candidates = [
            exe.parent / "../../../dist",
            exe.parent / "../../dist",
            Path("dist").resolve(),
            Path("../../dist").resolve(),
            Path("../../../dist").resolve(),
        ]
        d = Path("dist").resolve()
        for c in candidates:
            if c.exists() and c.is_dir():
                d = c.resolve()
                break

    if not d.exists():
        return []

    try:
        return [
            str(p) for p in d.iterdir()
            if p.is_file() and p.name != ".DS_Store"
        ]
    except Exception:
        return []

File: arm.64/langs/node/src/find.ts
import { readdirSync, statSync, existsSync } from 'node:fs';
import { resolve, join, dirname } from 'node:path';
import { fileURLToPath } from 'node:url';

const __dirname = dirname(fileURLToPath(import.meta.url));

export function find(targetDir?: string): string[] {
  let dir = targetDir;
  if (!dir) {
    if (process.env.DIST_DIR) {
      dir = process.env.DIST_DIR;
    } else {
      const candidates = [
        resolve(__dirname, '../../../dist'),
        resolve(__dirname, '../../dist'),
        resolve(process.cwd(), 'dist'),
        'dist',
      ];
      dir = candidates.find(c => existsSync(c)) || 'dist';
    }
  }

  if (!existsSync(dir)) return [];

  try {
    return readdirSync(dir)
      .map(name => join(dir!, name))
      .filter(p => statSync(p).isFile() && !p.endsWith('.DS_Store'));
  } catch {
    return [];
  }
}
